tapir/coop/models.py

```diff
@@ -719,3 +718,56 @@ def populate(
             old_frozen=old_frozen,
             new_frozen=new_frozen,
         )
+
+class MembershipResignation(models.Model):
+    share_owner = models.OneToOneField(
+        ShareOwner, on_delete=models.deletion.CASCADE, verbose_name=_("Shareowner"), related_name="share_owner",
+    )
+    cancellation_date = models.DateField(
+        default=timezone.now,
+        )
+    pay_out_day = models.DateField(null=True)
+    cancellation_reason = models.CharField(max_length = 1000)
+    coop_buys_shares_back = models.BooleanField()
+    willing_to_gift_shares_to_coop = models.BooleanField()
+    transfering_shares_to = models.OneToOneField(
+        ShareOwner, on_delete=models.deletion.PROTECT, verbose_name="OwnerToTransfer", null=True, related_name="owner_to_transfer",
+    )
+    paid_out = models.BooleanField(default=False)
+
+    class MembershipResignationQuerySet(models.QuerySet):
+        def with_name_or_id(self, search_string: str):
+            id_filter = (Q(share_owner__id__icontains=search_string))
```

Comment:
I think we want `Q(share_owner__id=search_string)` to check exact matches: if someone searches for member #123, then member #1234 should not show up.